docs/development/DuckTyping.md

```diff
@@ -491,6 +498,284 @@ AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
+
+In this section we provide some rules to apply. If you apply these consistently (and listen to the nullability warnings provided by the IDE and compiler) then you should hopefully avoid null reference exceptions in your integrations.
+
+This section assumes you're following the advice in [Choosing a proxy implementation](#choosing-a-proxy-implementation) and so always using either interface proxies or `[DuckCopy]` proxies 
+
+1. If you're using an interface proxy, derive the interface from `IDuckType`. This avoids the need for casting to check nullability.
+2. When duck chaining interface proxies, _always_ add `#nullable enable` to the file, and mark the interface property as nullable (using nullable reference types). e.g.
+```csharp
+#nullable enable // 👈 Always add this (to all new files)
+public interface IProxyMyHandler // The "root" proxy
+{
+    string Name { get; set; }
+    IProxyMyHandlerConfiguration? Configuration { get; } // duck-chained proxy
+                             // ☝ Marked nullable
+}
+```
+3. When duck chaining `[DuckCopy]` `struct`, _always_ mark the struct property as nullable. e.g.
+```csharp
+public interface IProxyMyHandler // The "root" proxy, doesn' matter what type it is
+{
+    string Name { get; set; }
+    ProxyMyHandlerConfiguration? Configuration { get; } // duck-chained [DuckCopy] struct proxy
+                             // ☝ Marked nullable
+}
+```
+4. _Before_ using _any_ ducktype proxy (from an integration or a duck-chained property), check it for null using the following helper methods:
+
+```csharp
+public class DuckType
+{
+    internal static bool IsNull<T>(T proxy);
+    internal static bool HasValue<T>(T proxy);
+}
+```
+
+For example:
+
+
+```csharp
+internal static CallTargetState OnMethodBegin<TTarget, TMyProxy>(TTarget instance, TMyProxy proxy)
+    where TMyProxy : IMyProxy, IDuckType
+{
+    if(DuckType.HasValue(proxy))
+    {
+        // safe to use proxy - underlying type is not null
+        if(DuckType.HasValue(proxy.Configuration)
+        {
+            // safe to use duck-chained property - underlying type is not null
+        }
+    }
+}
+```
+
+Yes this may be a bit annoying, but the consistently will make it easier to be confident about the nullability of the type, it helps the compiler with its nullability flow analysis, and it makes it easier to review. Note that it's important you also follow the other recommended steps however. 
+
+
+> Note, the above method has not yet been added. Added to the documentation for now to get feedback, and will implement and test in a subsequent PR if people agree. I intend to implement it as the following: 
+
+```csharp
+public class DuckType
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static bool IsNull<T>([NotNullWhen(false)] T proxy)
```

Comment:
Should we add a generic constraint that T implements IDuckType?

Reply:
No, because not all proxies _do_ implement `IDuckType` (e.g. `DuckCopy` structs) 😄  The goal with this was to have something that works for all cases.

I'm still on the fence as to whether this is worth it or not. I like the consistency, but it's a bit annoying if you know what you're doing 🤔